src/OrbitGl/App.cpp

```diff
@@ -798,21 +792,28 @@ ErrorMessageOr<void> OrbitApp::OnSaveCapture(const std::string& file_name) {
                                   timers_it_end);
 }
 
-void OrbitApp::OnLoadCapture(const std::string& file_name) {
-  CHECK(open_capture_callback_);
-  open_capture_callback_();
+Future<ErrorMessageOr<CaptureListener::CaptureOutcome>> OrbitApp::OnLoadCapture(
+    const std::string& file_name) {
   if (capture_window_ != nullptr) {
     capture_window_->set_draw_help(false);
   }
   ClearCapture();
   string_manager_->Clear();
-  thread_pool_->Schedule([this, file_name]() mutable {
+  auto load_future = thread_pool_->Schedule([this, file_name]() mutable {
     capture_loading_cancellation_requested_ = false;
-    capture_deserializer::Load(file_name, this, module_manager_.get(),
-                               &capture_loading_cancellation_requested_);
+
+    auto load_result = capture_deserializer::Load(file_name, this, module_manager_.get(),
+                                                  &capture_loading_cancellation_requested_);
+
+    if (load_result.has_value() && load_result.value() == CaptureOutcome::kComplete) {
```

Comment:
I don't quite get, why you check here if the result is already here and "complete" if you handle it the same on the caller, is this faster? If so, have you measured? Otherwise, I'd opt-in to reduce code-duplication and just return the future here and do keep the handling in the caller.

Reply:
Performance is not why I did that. 
The main reason is that the call to `OnCaptureComplete` (line 809) is a method in OrbitApp. 

This is my horrible attempt at visualizing the flow:
```
OrbitMainWindow::OpenCapture ->
    Show "Loading Progress Bar"
    OrbitApp::OnLoadCapture ->
        capture_deserializer::Load ->
           (....)
        OnCaptureComplete()
        return;
    Close "Loading Progress Bar"
    return;
```

What you suggest looks more like this
```
OrbitMainWindow::OpenCapture ->
    Show "Loading Progress Bar"
    OrbitApp::OnLoadCapture ->
        capture_deserializer::Load ->
           (....)
        return;
    app_->OnCaptureComplete()
    Close "Loading Progress Bar"
    return;
```

I dont know if any of this made sense. please let me know

Reply:
Right, I understand, but I still don't see what do we gain from this. Maybe we should discuss offline.

Reply:
As discussed offline, I basically was confused by the type of `load_result` (expected a future). Renaming the value or removing the `auto` should fix this.

Reply:
Changed the `auto` to the real type